presto-main/src/main/java/com/facebook/presto/sql/planner/EffectivePredicateExtractor.java

```diff
@@ -231,11 +231,12 @@ public Expression visitJoin(JoinNode node, Void context)
 
             switch (node.getType()) {
                 case INNER:
-                    return combineConjuncts(ImmutableList.<Expression>builder()
-                            .add(pullExpressionThroughSymbols(leftPredicate, node.getOutputSymbols()))
-                            .add(pullExpressionThroughSymbols(rightPredicate, node.getOutputSymbols()))
-                            .addAll(pullExpressionsThroughSymbols(joinConjuncts, node.getOutputSymbols()))
-                            .build());
+                    return pullExpressionThroughSymbols(combineConjuncts(ImmutableList.<Expression>builder()
+                            .add(leftPredicate)
+                            .add(rightPredicate)
+                            .add(combineConjuncts(joinConjuncts))
+                            .add(node.getFilter().orElse(TRUE_LITERAL))
+                            .build()), node.getOutputSymbols());
```

Comment:
I would extract variable out of:
```
combineConjuncts(ImmutableList.<Expression>builder()
                            .add(pullExpressionThroughSymbols(leftPredicate, node.getOutputSymbols()))	                            .add(leftPredicate)
                            .add(pullExpressionThroughSymbols(rightPredicate, node.getOutputSymbols()))	                            .add(rightPredicate)
                            .addAll(pullExpressionsThroughSymbols(joinConjuncts, node.getOutputSymbols()))	                            .add(combineConjuncts(joinConjuncts))
                            .build());	                            .add(node.getFilter().orElse(TRUE_LITERAL))
                            .build()
```

Reply:
Something is wrong with that code snippet.